cdap-api/src/main/java/co/cask/cdap/api/metadata/MetadataWriterContext.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.metadata;
+
+import java.util.Map;
+import javax.ws.rs.NotFoundException;
+
+/**
+ * The context for emitting metadata from programs
+ */
+public interface MetadataWriterContext {
+
+  /**
+   * Adds the specified {@link Map} to the metadata of the specified {@link MetadataEntity metadataEntity}.
+   * Existing keys are updated with new values, newer keys are appended to the metadata.
```

Comment:
What do mean by newer keys are appended?? I though properties is just a `Map<String, String>`?